src/auth/client.ts

```diff
@@ -26,6 +26,10 @@ export interface AuthenticateServerOptions extends AbortOptions {
   hostname?: string
 }
 
+export interface VerifyPeerOptions {
+  verifyPeer?(peerId: PeerId, abort?: AbortSignal): Promise<boolean>
```

Comment:
For forwards compatibility this is better as an `AbortOptions` as it lets us pass additional options in later without a breaking change.

```suggestion
  verifyPeer?(peerId: PeerId, options?: AbortOptions): Promise<boolean>
```